### PR TITLE
Id128 fixes and more

### PR DESCRIPTION
**Pull request by poettering**
Adds two more doc fixes, as suggested by @evverx and @gdamjan.

### CODE REVIEW

File: CODING_STYLE [hunk 1]
```diff
@@ -406,3 +406,26 @@
   shorts as their name would suggest, but on uint32_t and uint16_t. Also,
   "network byte order" is just a weird name for "big endian", hence we might
   want to call it "big endian" right-away.
+
+- You might wonder what kind of common code belongs in src/shared/ and what
+  belongs in src/util/. The split is like this: anything that uses public APIs
```
Comment (evverx) [inline CODING_STYLE:411]: I think should be `src/basic` (instead of `src/util`)

File: CODING_STYLE [hunk 2]
```diff
@@ -406,3 +406,26 @@
...
+  from src/basic, but not from src/shared/. To summarize:
+
+  src/basic/      → may be used by all code in the tree
+                  → may not use any code outside of src/basic/
+
+  src/shared/     → may be used by all code in the tree, except for code in src/basic/
+                  → may not use any code outside of src/basic/, src/shared/, src/libsystemd/
+
+  src/libsystemd/ → may be used by all code in the tree, except for code in src/basic/
+                  → may not use any code outside of src/basic/, src/shared/, src/libsystemd/
```
Comment (evverx) [inline CODING_STYLE:425]: `src/libsystemd` may not use any code outside of `src/basic/`, ~~src/shared/~~, `src/libsystemd/`

File: CODING_STYLE (hunk 1 above)
Comment (poettering) [inline CODING_STYLE:411]: indeed

File: CODING_STYLE [hunk 3]
```diff
@@ -406,3 +406,26 @@
...
+  we expose (i.e. any of the sd-bus, sd-login, sd-id128, ... APIs) must be
+  located in src/shared/. All stuff that only uses external libraries from
+  other projects (such as glibc's APIs), or APIs from src/basic/ itself should
+  be placed in src/basic/. Conversely, src/libsystemd/ may only use symbols
+  from src/basic, but not from src/shared/. To summarize:
+
+  src/basic/      → may be used by all code in the tree
+                  → may not use any code outside of src/basic/
+
+  src/shared/     → may be used by all code in the tree, except for code in src/basic/
```
Comment (evverx) [inline CODING_STYLE:421]: may be used by all code in the tree, except for code in `src/basic/`, **`src/libsystemd`**

File: CODING_STYLE (hunk 2 above)
Comment (poettering) [inline CODING_STYLE:425]: indeed

File: CODING_STYLE [hunk 4]
```diff
@@ -406,3 +406,26 @@
...
+- You might wonder what kind of common code belongs in src/shared/ and what
+  belongs in src/util/. The split is like this: anything that uses public APIs
+  we expose (i.e. any of the sd-bus, sd-login, sd-id128, ... APIs) must be
+  located in src/shared/. All stuff that only uses external libraries from
+  other projects (such as glibc's APIs), or APIs from src/basic/ itself should
+  be placed in src/basic/. Conversely, src/libsystemd/ may only use symbols
+  from src/basic, but not from src/shared/. To summarize:
+
+  src/basic/      → may be used by all code in the tree
+                  → may not use any code outside of src/basic/
```
Comment (fsateler) [inline CODING_STYLE:419]: Hmm. 

```
% git grep -i sd'[-_]' src/basic/
src/basic/btrfs-util.h:#include "sd-id128.h"
src/basic/btrfs-util.h:        sd_id128_t uuid;
src/basic/btrfs-util.h:        sd_id128_t parent_uuid;
src/basic/hashmap.c:/* Tracks all existing hashmaps. Get at it from gdb. See sd_dump_hashmaps.py */
src/basic/hashmap.h: * - store extra data for debugging and statistics (see tools/gdb-sd_dump_hashmaps.py)
src/basic/log.c:#include "sd-messages.h"
src/basic/log.c:                                LOG_MESSAGE_ID(SD_MESSAGE_INVALID_CONFIGURATION),
src/basic/log.c:                                LOG_MESSAGE_ID(SD_MESSAGE_INVALID_CONFIGURATION),
src/basic/log.h:#include "sd-id128.h"
src/basic/log.h:#define LOG_MESSAGE_ID(x) "MESSAGE_ID=" SD_ID128_FORMAT_STR, SD_ID128_FORMAT_VAL(x)
src/basic/memfd-util.c:                        g = strappend("sd-", e);
```

so log and btrfs-util use code from src/systemd

Comment (poettering) [inline CODING_STYLE:419]: yeah, it's a bit ugly, but if you look closely you'll see they only use types and macros from those two headers, not actual symbols. I looked into removing those usages too, but it didn't appear worth it, as the changes would be quite large.

**Pull request merged**